packages/vite/src/node/optimizer/index.ts

```diff
@@ -79,6 +79,12 @@ export interface DepOptimizationOptions {
    * @deprecated use `esbuildOptions.keepNames`
    */
   keepNames?: boolean
+  /**
+   * List of file extensions that can be optimized. A corresponding esbuild
+   * plugin must exist to handle the specific extension.
+   * @experimental
+   */
+  supportedExtensions?: string[]
```

Comment:
Marked this as experimental so it's not exposed in the type def. In the future, Vue / Marko could implement the same (?), if the DSL can be compiled to JS directly.

Reply:
That could work too, but we might need a default set of extensions then so it aligns with `resolve.extensions`. Currently the extensions are in a [regex](https://github.com/vitejs/vite/blob/f63a72e77214ea31884680a3c5facf4fcee5d226/packages/vite/src/node/constants.ts#L20) so I might have to expand that first. The idea of `supportedExtensions` was more of an additive option on top of the regex.

Reply:
Pushed and renamed this to `extensions`. I've decided to not have the default list as `.mjs`, `.js`, and `.ts`, since it's very unlikely someone wants to remove one of the extensions, whereas `resolve.extensions` it's plausible. Instead this option is additive to the default list, sort-of how `resolve.conditions` work.